esp/services/esdl_svc_engine/esdl_store.cpp

```diff
@@ -42,24 +42,57 @@ extern bool trimXPathToParentSDSElement(const char *element, const char * xpath,
 
 class CEsdlSDSStore : implements IEsdlStore, public CInterface
 {
+private:
+    bool m_isAttached;
+    CriticalSection m_attachCrit;
 public:
     IMPLEMENT_IINTERFACE;
 
     CEsdlSDSStore()
     {
         ensureSDSPath(ESDL_DEFS_ROOT_PATH);
         ensureSDSPath(ESDL_BINDINGS_ROOT_PATH);
+        m_isAttached = true;
     }
     virtual ~CEsdlSDSStore() { }
 
+    virtual void detachFromBackend()
+    {
+        CriticalBlock cb(m_attachCrit);
+        m_isAttached = false;
+    }
+
+    virtual void attachToBackend()
+    {
+        CriticalBlock cb(m_attachCrit);
+        m_isAttached = true;
+    }
+
+    virtual bool isAttachedToBackend()
+    {
+        CriticalBlock cb(m_attachCrit);
+        return m_isAttached;
+    }
+
+    ISDSManager& QuerySDS()
```

Comment:
Minor, but most conventions use lower case initial letter on method name

Reply:
Changed method name to checkQuerySDS(), please review again @rpastrana